app/reducers.js

```diff
@@ -5,20 +5,18 @@
 import { combineReducers } from 'redux-immutable';
 import { connectRouter } from 'connected-react-router/immutable';
 
-import history from 'utils/history';
 import globalReducer from 'containers/App/reducer';
 import languageProviderReducer from 'containers/LanguageProvider/reducer';
 
 /**
  * Merges the main reducer with the router state and dynamically injected reducers
  */
 export default function createReducer(injectedReducers = {}) {
-  const rootReducer = combineReducers({
-    global: globalReducer,
-    language: languageProviderReducer,
-    router: connectRouter(history),
-    ...injectedReducers,
-  });
-
-  return rootReducer;
+  return history =>
+    combineReducers({
+      router: connectRouter(history),
+      global: globalReducer,
+      language: languageProviderReducer,
+      ...injectedReducers,
+    });
```

Comment:
I didn't understand why `createReducer` needs to rerun `connectRouter` with the `history` each time. Would you mind clarifying that please?

And if it's necessary wouldn't it be easier if `history` was just a second argument of `createReducer`? So that you only call one function. At first look, this seems more convoluted than it needs to be.

Reply:
I just followed the example and recommendations at connected-react-router for this change.